avocado/utils/ssh.py

```diff
@@ -248,3 +249,20 @@ def copy_files(self, source, destination, recursive=False):
             return result.exit_status == 0
         except process.CmdError as exc:
             raise NWException("failed to copy file {}".format(exc))
+
+    def remove_control_path(self):
+        """
+        Check for ssh connection if failed remove control path file
+        and reconnect it.
+
+        :returns: True if success or else raise exception
+        :rtype: bool
+        """
+        if not self.connect():
```

Comment:
I'm trying to understand why you need to establish a connection to the remote host, in order to remove a local file.

self.connect() is not only checking for connection but is trying to connect to the  host. And this is not what your method or docstring is saying.

Reply:
actually this function is for, if check connection failed than we will got for quit() to close session if that also not work than we will remove the control path file.if you want me to do this method for only removing control path file thats also fine.

Reply:
@smruti77 I'm just trying to understand what you are trying to do here. When you say on docstring " Check for ssh connection if failed ..." I was reading "Check for ssh connection" (just a check), but you are actually trying to connect. 

Could you please try to improve the docstring (can be here on a comment) with a very detailed docstring of what are you trying to achieve here? Maybe the sentence is not correct and I'm getting confused.

Reply:
ok